dspace/config/modules/assetstore.cfg

```diff
@@ -0,0 +1,68 @@
+#---------------------------------------------------------------#
+#-----------------STORAGE CONFIGURATIONS------------------------#
+#---------------------------------------------------------------#
+# Configuration properties used by the bitstore.xml config file #
+#                                                               #
+#---------------------------------------------------------------#
+
+# assetstore.dir, look at DSPACE/config/spring/api/bitstore.xml for more options
+assetstore.dir = ${dspace.dir}/assetstore
+
+# Configures the primary store to be local or S3.
+# This value will be used as `incoming` default store inside the `bitstore.xml`
+# Possible values are:
+#     - 0: to use the `localStore`;
+#     - 1: to use the `s3Store`. 
+# If you want to add additional assetstores, they must be added to that bitstore.xml
+# and new values should be provided as key-value pairs in the `stores` map of the
+# `bitstore.xml` configuration. 
+assetstore.index.primary = 0
+
+# This secondary store is used as backup store when
+# creating new bitstreams and the primary store is failing
+# its initialization process.
+# If you are using the `s3Store` as primary is convenient to use
+# a secondary like the `localStore`, in this way if the amazon one will
+# fail during its initialization, the secondary will be used for the new bitstreams.
+# Possible values are:
+#     - 0: to use the `localStore`;
+#     - 1: to use the `s3Store`.
+assetstore.index.secondary = 0
```

Comment:
@vins01-4science : I feel like this is "scope creep", as we have not previously discussed the need for a "secondary" store location. I'm not convinced that it is necessary for most sites & I worry it'd just add more confusion to our storage configuration.

I'm also worried this new feature will cause unexpected storage issues if content is stored in the wrong storage location without your realizing it.

Here's an extreme example which is now problematic:
* Suppose you want to use `s3Store` because you don't have much local storage (100GB), but you have a `localStore` backup just in case.
* Suppose Amazon S3 is temporarily down while you are importing a large amount of content (500GB)
* Based on this implementation, you won't receive an error letting you know this large upload cannot be done **until you fill up all your local storage**.  Essentially, the secondary store will be used automatically for imported content, even if you don't want any of your content stored in that secondary store.  And there's no way to then move already imported content easily from your `localStore` to `s3Store` to free up your local storage.

I'm not in favor of a secondary store location being added at this time.  I think it'd require more discussion.   Could we just remove this secondary store option?  It's not something we asked for and it's not in the original goals of this PR....so it feels like it belongs in a separate PR.

Reply:
@tdonohue : Sure, I will remove it, but if the `s3StoreService` isn't configured correctly, when someone will switch from `local` to `s3` some bitstream could be lost.

Reply:
@vins01-4science : If `s3StoreService` is misconfigured, it'll throw an error. I think that's appropriate here, because it blocks all new bitstreams from being added until the storage settings are fixed.

Personally, I feel most people will want to know if their storage is misconfigured (by seeing the thrown error) instead of DSpace silently hiding the misconfiguration & storing the bitstreams in the "secondary" location.

I could be wrong about my view here, but as I said, I think this needs more discussion. I'm not comfortable with adding this feature into 7.4 at the last moment as we haven't talked about it at all previously.

Reply:
@tdonohue : Yes, sure I will remove it.
Thank you.

Reply:
@tdonohue : removed that configuration and behavior.
Thank you.

Reply:
I have some ideas about policy-directed bitstore selection (for replication, nearline storage of huge objects, etc.) and a "secondary store" policy could be part of it, if we ever do that.